hooks/basic/collector.py

```diff
@@ -357,15 +416,25 @@ def _add_file_item(self, settings, parent_item, path, is_sequence=False, seq_fil
         if is_sequence:
             properties["sequence_paths"] = seq_files
 
+        if work_path_template:
+            properties["work_path_template"] = work_path_template
+
+        # build the context of the item
+        context = self._get_item_context_from_path(properties, path)
+
         # create and populate the item
         file_item = parent_item.create_item(
             item_type,
             type_display,
             display_name,
             collector=self.plugin,
+            context=context,
             properties=properties
         )
 
+        # resolve work_path_template for the item
+        file_item.properties["work_path_template"] = self._resolve_work_path_template(file_item.properties, path)
+
```

Comment:
Please see comment above for why this got moved to on_context_changed(). Just so you are aware, the `work_path_template` returned at this point in the evaluation will return the environment configuration for the application context rather than the item's context, so this may not be what you want.

Reply:
Ah, I forgot to remove it, this is already being evaluated in on_context_changed.